File: domain/src/main/java/io/prizy/domain/contest/ports/ContestPublisher.java
package io.prizy.domain.contest.ports;

import io.prizy.domain.contest.event.ContestCreated;
import io.prizy.domain.contest.event.ContestUpdated;

/**
 *  @author Nidhal Dogga
 *  @created 4/24/2022 1:50 AM
 */

public interface ContestPublisher {
  void publish(ContestCreated event);
  void publish(ContestUpdated event);
}

File: domain/src/main/kotlin/io/prizy/domain/contest/ports/ContestPublisher.kt
package io.prizy.domain.contest.ports

import io.prizy.domain.contest.event.ContestCreated
import io.prizy.domain.contest.event.ContestUpdated

/**
 *  @author Nidhal Dogga
 *  @created 4/24/2022 1:50 AM
 */

interface ContestPublisher {
  suspend fun publish(event: ContestCreated)
  suspend fun publish(event: ContestUpdated)
}
